common/src/chain/transaction/output/htlc.rs

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2024 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// TODO: consider removing this in the future when fixed-hash fixes this problem
+#![allow(clippy::non_canonical_clone_impl)]
+
+use std::time::Duration;
+
+use serialization::{Decode, Encode};
+use zeroize::Zeroize;
+
+use super::Destination;
+
+#[derive(Debug, Clone, PartialEq, Eq, Encode, Decode, serde::Serialize, serde::Deserialize)]
+pub struct HashedTimelockContract {
+    // can be spend either by a specific address that knows the secret
+    pub secret_hash: HtlcSecretHash,
+    pub spend_key: Destination,
+
+    // or by a multisig after timelock expires making it possible to refund
+    pub refund_timelock: Duration,
+    pub refund_key: Destination,
+}
+
+#[derive(Zeroize, Clone, Encode, Decode, PartialEq, Eq, Debug)]
+pub struct HtlcSecret {
+    pub secret: [u8; 20],
+}
+
+fixed_hash::construct_fixed_hash! {
+    #[derive(Encode, Decode)]
+    pub struct HtlcSecretHash(20);
```

Comment:
Why 20, not 32?

Reply:
I believe because 20 is enough in this case. It's not me who come up with this number:
https://lightning.network/lightning-network-paper.pdf#page=30

Reply:
I mean the size of the secret should be 32 but the resulting hash is 20

Reply:
I'm OK with that.

Reply:
done